exporter/exporter.go

```diff
@@ -62,77 +66,123 @@ func New(opts *Opts) (*Exporter, error) {
 		opts = new(Opts)
 	}
 
-	client, err := connect(context.Background(), opts.URI)
-	if err != nil {
-		return nil, err
-	}
-
 	if opts.Logger == nil {
 		opts.Logger = logrus.New()
 	}
 
+	ctx := context.Background()
+	var client *mongo.Client
+	var err error
+	// Use global connection pool.
+	if opts.GlobalConnPool {
+		client, err = connect(ctx, opts.URI)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	exp := &Exporter{
 		client:           client,
-		collectors:       make([]prometheus.Collector, 0),
 		path:             opts.Path,
 		logger:           opts.Logger,
+		opts:             opts,
 		webListenAddress: opts.WebListenAddress,
 	}
 
-	if len(opts.CollStatsCollections) > 0 {
-		exp.collectors = append(exp.collectors, &collstatsCollector{
+	return exp, nil
+}
+
+func (e *Exporter) makeRegistry(ctx context.Context, client *mongo.Client) *prometheus.Registry {
+	registry := prometheus.NewRegistry()
```

Comment:
Can we use NewPedanticRegistry there?

Reply:
@askomorokhov Please actually check that NewPedanticRegistry does not explode there – because it can